src/app/core/tasks/threadpool.nim

```diff
@@ -1,260 +1,45 @@
-            threadsIdle.add (notification.id, tpl.thr, tpl.chanSendToTask)
-            trace "[threadpool] removing from threadsBusy",
-              newlength=(threadsBusy.len - 1), threadid=notification.id
-            threadsBusy.del notification.id
-
-            if taskQueue.len > 0:
-              trace "[threadpool] removing from taskQueue",
-                newlength=(taskQueue.len - 1)
-              task = taskQueue[0]
-              taskQueue.delete 0, 0
-
-              trace "[threadpool] removing from threadsIdle",
-                newlength=(threadsIdle.len - 1)
-              let tpl = threadsIdle[0]
-              threadsIdle.delete 0, 0
-              trace "[threadpool] adding to threadsBusy",
-                newlength=(threadsBusy.len + 1), threadid=tpl.id
-              threadsBusy.add tpl.id, (tpl.thr, tpl.chanSendToTask)
-              await tpl.chanSendToTask.send(task.safe)
-
-          else:
-            error "[threadpool] unknown notification", notice=notification.notice
-        except Exception as e:
-          warn "[threadpool] unknown error in thread notification", message=task, error=e.msg
-
-      else: # must be a request to do task work
-        if allReady < arg.size or threadsBusy.len == arg.size:
-          # add to queue
-          trace "[threadpool] adding to taskQueue",
-            newlength=(taskQueue.len + 1)
-          taskQueue.add task
-
-        # do we have available threads in the threadpool?
-        elif threadsBusy.len < arg.size:
-          # check if we have tasks waiting on queue
-          if taskQueue.len > 0:
-            # remove first element from the task queue
-            trace "[threadpool] adding to taskQueue",
-              newlength=(taskQueue.len + 1)
-            taskQueue.add task
-            trace "[threadpool] removing from taskQueue",
-              newlength=(taskQueue.len - 1)
-            task = taskQueue[0]
-            taskQueue.delete 0, 0
-
-          trace "[threadpool] removing from threadsIdle",
-            newlength=(threadsIdle.len - 1)
-          let tpl = threadsIdle[0]
-          threadsIdle.delete 0, 0
-          trace "[threadpool] adding to threadsBusy",
-            newlength=(threadsBusy.len + 1), threadid=tpl.id
-          threadsBusy.add tpl.id, (tpl.thr, tpl.chanSendToTask)
-          await tpl.chanSendToTask.send(task.safe)
-
-  var allTaskThreads = newSeq[tuple[id: int, thr: Thread[TaskThreadArg]]]()
-
-  for tpl in threadsIdle:
-    tpl.chanSendToTask.close()
-    allTaskThreads.add (tpl.id, tpl.thr)
-  for id, tpl in threadsBusy.pairs:
-    tpl.chanSendToTask.close()
-    allTaskThreads.add (id, tpl.thr)
-
-  chanSendToMain.close()
-  chanRecvFromMainOrTask.close()
-
-  trace "[threadpool] waiting for all task threads to stop"
-  for tpl in allTaskThreads:
-    debug "[threadpool] join thread", threadid=tpl.id
-    joinThread(tpl.thr)
+      error "[threadpool task thread] exception", error=e.msg
+  except Exception as e:
```

Comment:
In general, it's not advised to `except Exception`, as it may or may not catch `Defect` as well (which is derived from `Exception`). You'd instead want:
```suggestion
  except CatchableError as e:
```
However, this of course depends on how exceptions are handled upstack.

As a general comment on these two try/excepts... it might be considered to send caught exceptions back to the calling thread in the shape of an error, which the UI could react to if needed. 

More info can be found in the [Exceptions section of the Nim style guide](https://status-im.github.io/nim-style-guide/errors.exceptions.html).